design/ceph/resource-dependencies.md

```diff
@@ -0,0 +1,318 @@
+- CephObjectStoreUsers in the same namespace that have `spec.store == <CephObjectStore.metadata.name>`
+
+Dependents via lib-bucket-provisioner:
+- ObjectBucketClaims that have:
+  - `spec.endpoint.bucketHost == <provider's service name>.<ceph cluster namespace>.svc` AND
+  - `spec.endpoint.bucketName == <a provider bucket>`
+
+Dependents via COSI:
+- Buckets with a reference to a provider's service and bucket (specific paths TBD by COSI design)
+
+#### `CephObjectZone`, `CephObjectZoneGroup`, and `CephObjectRealm`
+These resources are all part of Rook-Ceph's multi-site object storage strategy.
+
+`CephObjectZone` creates pools. A zone can be effectively thought of as the "object store" itself.
+Dependents which can consume this provider's pools:
+- CephObjectStores in the same namespace that have `spec.zone.name == CephObjectZone.metadata.name`
+- CephNFS in the same namespace that have `spec.pool = <a provider pool>`
+- CephClient in the same namespace that has an `spec.caps` value with the string `pool=<a provider pool>`
+
+`CephObjectRealm` has dependents:
+- CephObjectZoneGroups in the same namespace that have `spec.realm == CephObjectRealm.metadata.name`
+
+`CephObjectZoneGroup` has dependents:
+- CephObjectZones in the same namespace that have `spec.zoneGroup == CephObjectZoneGroup.metadata.name`
+
+#### Mirroring resources
+CephRBDMirror has dependents:
+- CephBlockPools in the same namespace that have `spec.mirroring.enabled == true`
+
+CephFilesystemMirror has dependents:
+- CephFilesystems in the same namespace that have `spec.mirroring.enabled == true`
+
+
+## Analysis and discussion
+### Reusable patterns
+We can identify some common metrics for determining whether a resource is a dependent of a given
+"provider" resource. Not all metrics are always applicable, but each these metrics appear more than
+once. It should be possible to design reusable patterns/methods for reusing logic.
+- Is a dependent in the same namespace as the provider?
+- Does a dependent reference the current Rook-Ceph operator?
+- Does a dependent reference the Ceph cluster namespace of the provider?
+- Does the dependent reference pools created by provider?
+- Does the dependent reference the provider by name? (e.g., CephObjectZone references CephObjectZoneGroup)
+
+### User feedback
+It will be important for the user to understand why resources are not being deleted if Rook is
+blocking the deletion. This design proposes that the Rook operator report to the user when it is
+blocking deletion of a resource due to dependents in two ways:
+- report it in a status condition of the resource object being deleted
+- report a Kubernetes Event ([reference](https://kubernetes.io/blog/2018/01/reporting-errors-using-kubernetes-events/))
+
+Using both of these methods will maximize user visibility.
+
+**Status:**
+Reported statuses will be modified as follows:
+1. Object's status.phase should be changed to "Deleting" as soon as a deletion timestamp is detected
```

Comment:
a `kubectl get <CR>` should already show it deleted, but updating the phase certainly wouldn't hurt if we're already reporting conditions and events